cinch/bin/wrappers.py

```diff
@@ -13,30 +15,115 @@
     ansible = local['ansible-playbook']
+    command_handler(ansible, ansible_args)
+
+
+def call_linchpin(work_dir, *args):
+    """
+    Wraps a call out to the linchpin executable, and then kicks off a cinch
+    Ansible playbook if necessary.
+
+    :param work_dir: The linch-pin working directory that contains a PinFile
+    and associated configuration files
+    :param args: An array of other command-line arguments to linchpin to pass
+    :return: The exit code returned from linchpin, or 255 if errors come from
+    elsewhere
+    """
+    # Construct the arguments to pass to linchpin by munging the arguments
+    # provided to this method
+    linchpin_args = list()
+    linchpin_args.extend(args)
+    # cinch will only support a subset of linchpin subcommands
+    supported_cmds = ['rise']
+    for cmd in linchpin_args[0]:
+        if cmd not in supported_cmds:
+            raise Exception('linchpin command \'' + cmd + '\' not '
+                            'supported by cinch')
+    # Attempt to open the linch-pin PinFile
+    try:
+        with open(path.join(work_dir, 'PinFile')) as pin_file:
+            pin_file_yaml = yaml.safe_load(pin_file)
+    except IOError:
+        print('linch-pin PinFile not found in ' + work_dir)
+        sys.exit(1)
+    # We must find a topology section named 'cinch' to determine where our
+    # inventory file will live
+    try:
+        cinch_topology = 'cinch'
+        topology = pin_file_yaml[cinch_topology]['topology']
+    except KeyError:
+        print('linch-pin PinFile must contain a topology '
+              'section named \'' + cinch_topology + '\'')
+        sys.exit(1)
+    #  The inventory file generated by linchpin that will be used by cinch for
+    #  configuration
+    try:
+        topology_path = path.join(work_dir, 'topologies', topology)
+        with open(topology_path) as topology_file:
+            topology_yaml = yaml.safe_load(topology_file)
+    except IOError:
+        print('linch-pin topology file not found at ' +
+              topology_path)
+        sys.exit(1)
+    inventory_file = topology_yaml['topology_name'] + '.inventory'
+
+    # As of right now the linch-pin working directory is not configurable, so
+    # we essentially have a 'pushd' here to switch to the linch-pin working
+    # directory as needed. This code can be removed if the following issue is
+    # resolved:
+    # https://github.com/CentOS-PaaS-SIG/linch-pin/issues/119
+    @contextmanager
+    def pushd(new_dir):
+        previous_dir = getcwd()
+        chdir(new_dir)
+        yield
+        chdir(previous_dir)
+
+    with pushd(work_dir):
+        linchpin = local['linchpin']
+        exit_code = command_handler(linchpin, linchpin_args)
+    # If linchpin is asked to provision resources, we will then run our
+    # cinch playbooks
+    # TODO: Add support for 'drop' and other supported linchpin command
+    # subsets
+    if 'rise' in linchpin_args[0] and exit_code == 0:
```

Comment:
Do you want to use "in" instead of "==" for the operator, here? "in" will return true for other strings there such as "arisen" or such. Not that those are valid linchpin commands, but in the future they might be.